### ISSUE TITLE
Code navigation and syntax highlighting in MSVS breaks after switching Git branches

### ISSUE THREAD

**Issue by atsapura** [dotnet/fsharp#13496]
Sometimes (not always, unfortunately) syntax highlighting and navigation breaks in some tabs, not all of them though. 

**Repro steps**

1. Have a solution open, multiple tabs open, some of them pinned (not sure if it matters). In my case, a lot of tabs were open.
2. Switch branch, I suspect, it's better if it leads to projects or solution being reloaded.
3. Hope that you caught the issue this time.

So far I haven't been able to reproduce this but via `File -> Close Solution -> Reopen Solution`, but who knows.

**Expected behavior**

This doesn't happen, code navigation and highlighting remain intact.

**Known workarounds**

Reopen Visual Studio. 

**Related information**

* Operating system: Win 10
* .NET Runtime kind: dotnet 6.0
* MSVS version: 17.2.5, but it was definitely present in some of the earlier versions as well.

**Comment by vzarytovskii** [comment]
> I have exactly the same issue in Rider.
> 
> * Operating system: Win 11
> * .NET Runtime kind: dotnet 6.0
> * Rider version: 2022.1.1 & 2022.1.2

Rider should probably be reported to the JB's YouTrack.

**Comment by mvsmal** [comment]
@vzarytovskii @auduchinok Will do. 

Just thought since it looks like exactly the same issue it might be somewhere in a shared place (FCS?). Just a wild guess, I have no idea what VS and Rider have in common in regards to F#.

**Comment by vzarytovskii** [comment]
> @vzarytovskii @auduchinok Will do. 
> 
> Just thought since it looks like exactly the same issue it might be somewhere in a shared place (FCS?). Just a wild guess, I have no idea what VS and Rider have in common in regards to F#.

Yeah, in vs it's most likely project system or us not kicking in re-check.

**Comment by atsapura** [comment]
UPD:
Just caught it once more and this time 1 file got stuck so hard, that even reopening VS didn't solve it. Clean/Rebuild didn't solve it. The only thing that did fix it was manual deleting of all the `bin` and `obj` folders.
